server/src/test/java/org/opensearch/wlm/cancellation/DefaultTaskSelectionStrategyTests.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.wlm.cancellation;
+
+import org.opensearch.action.search.SearchAction;
+import org.opensearch.action.search.SearchTask;
+import org.opensearch.cluster.metadata.QueryGroup;
+import org.opensearch.core.tasks.TaskId;
+import org.opensearch.core.tasks.resourcetracker.ResourceStats;
+import org.opensearch.core.tasks.resourcetracker.ResourceStatsType;
+import org.opensearch.core.tasks.resourcetracker.ResourceUsageMetric;
+import org.opensearch.search.ResourceType;
+import org.opensearch.tasks.Task;
+import org.opensearch.tasks.TaskCancellation;
+import org.opensearch.test.OpenSearchTestCase;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+public class DefaultTaskSelectionStrategyTests extends OpenSearchTestCase {
```

Comment:
We should add at least one test where the Strategy is returning non empty set of tasks from given list of tasks.

Reply:
@kaushalmahi12 the test - `testSelectTasksToCancelSelectsTasksMeetingThreshold_ifReduceByIsGreaterThanZero` is returning a non empty set of selected tasks

Reply:
I see! fair enough.